packages/fxa-shared/db/redis.ts

```diff
@@ -188,83 +192,138 @@ export class RedisShared {
     const p1 = this.redis.pruneSessionTokens(uid, JSON.stringify(tokenIds));
     const p2 = this.rejectInMs(p1, this.timeoutMs);
-    return Promise.race([p1, p2]);
+    const result = Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   async pruneRefreshTokens(
     uid: Buffer | String,
     tokenIdsToPrune: Buffer[] | string[]
   ) {
+    this.metrics?.increment('redis.pruneRefreshTokens');
+    const span = tracer.startSpan('redis.pruneRefreshTokens');
     const p1 = this.redis.hdel(hex(uid), ...tokenIdsToPrune.map((v) => hex(v)));
     const p2 = this.resolveInMs(p1, this.timeoutMs);
-    return await Promise.race([p1, p2]);
+    const result = await Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   async getSessionTokens(uid: string) {
+    this.metrics?.increment('redis.getSessionTokens');
+    const span = tracer.startSpan('redis.getSessionTokens');
     try {
       const p1 = this.redis.getSessionTokens(uid);
       const p2 = this.rejectInMs(p1, this.timeoutMs);
       const value = await Promise.race([p1, p2]);
+
+      if (value?.length > 0) {
+        span.setAttribute('tokens.length', value.length);
+        this.metrics?.histogram(
+          'redis.getSessionTokens.tokens.length',
+          value.length
+        );
+      }
       return JSON.parse(value as string);
     } catch (e) {
       this.log?.error('RedisShared', {
         error: e,
       });
       return {};
+    } finally {
+      span.end();
     }
   }
 
   async getAccessTokens(uid: Buffer | String) {
+    this.metrics?.increment('redis.getAccessTokens');
+    const span = tracer.startSpan('redis.getAccessTokens');
     try {
       const values = await this.redis.getAccessTokens(hex(uid));
+
+      if (values?.length) {
+        span.setAttribute('tokens.length', values.length);
+        this.metrics?.histogram(
+          'redis.getAccessTokens.tokens.length',
+          values.length
+        );
+      }
       return values.map((v: string) => AccessToken.parse(v));
     } catch (e) {
       this.log?.error('RedisShared', {
         error: e,
       });
       return [];
+    } finally {
+      span.end();
     }
   }
 
   async getAccessToken(uid: Buffer | String) {
+    this.metrics?.increment('redis.getAccessToken');
+    const span = tracer.startSpan('redis.getAccessToken');
     try {
       const value = await this.redis.getAccessToken(hex(uid));
       if (value) return AccessToken.parse(value);
     } catch (e) {
       this.log?.error('RedisShared', {
         error: e,
       });
+    } finally {
+      span.end();
     }
 
     return null;
   }
 
   async touchSessionToken(uid: string, token: any) {
+    this.metrics?.increment('redis.touchSessionToken');
+    const span = tracer.startSpan('redis.touchSessionToken');
     // remove keys with null values
     const json = JSON.stringify(token, (k, v) => (v == null ? undefined : v));
+    span.setAttribute('touchSessionToken.json.size', json.length);
```

Comment:
Not sure of any of these matter, but should this have a `redis.` prefix?
I noticed other usages above (apart from the other 2 I commented on above) are prefixed by `redis.` and generally match the `this.metrics?.increment()` and `tracer.startSpan()` labels.

Reply:
I like the recommendation just so things are consistent. The attributes are attached to a span, so a unique name isn’t critical. That being said, you can search/filter by attribute name, so it is probably better for it to be specific.